src/common/codeUtil.ts

```diff
@@ -3,7 +3,19 @@ export class CodeUtil {
      * trim unfoucs code.
      * @param origin any string
      */
-    public static purity(origin: string): string {
+    public static purity_greedy(origin: string): string {
+        if (!origin) return '';
+        // TODO: untest
+        return origin
+            .replace(/;.+/, '')
+            .replace(/".*?"/g, '')
+            .replace(/\{.*\}/g, '')
+            .replace(/ +/g, ' ')
+            .replace(/\bgui\b.*/gi, '')
+            .replace(/\b(msgbox)\b.+?%/gi, '$1');
+    }
+
+    public static purity(origin: string, greedy: boolean = false): string {
```

Comment:
I added that greedy parameter, and decided against it, because this is called EVERY line, and if statement is expensive,

what would be cool would be to replace `purity` with `purity_greedy` everywhere, but idk if it's a breaking change

I forgot to remove that unused parameter

Reply:
It shouldn't be a breaking change. What is the difference between `purity_greedy` and `purity`? I see none. Just use `purity` since we now have two functions doing the same thing.

Reply:
it really isn't clear. it doesn't even show it.
line 12
`            .replace(/\{.*\}/g, '')`
line 24
`            .replace(/\{.*?\}/g, '')`

Reply:
I noticed that. Removing the `?` from the regex does make it appear simpler, but they have the same functionality. Since the `?` follows a `*`, it never does anything. The `*` following the `.` captures 0 or more instances of any char and the `?` captures 0 or 1 instance(s) of any char. With or without the `?`, the regex captures 0 or more instances of any char.

If I'm wrong, please provide a counterexample. For simplicity, consider `be*?d` and note that it captures:
- `bd`
- `bed`
- `beed`

Now consider `be*d`. It captures the exact same things.

Reply:
I recommend you go ahead and replace all `*?` with `*` for simplicity. I can manually test the functionalities that depend on `purity` if you don't feel comfortable doing it.

Reply:
regex lesson: normally ? means 0 or 1 instance(s), but when ? is placed after *, it transforms the * into lazy, because * is greedy by Default

you can use https://regexr.com/ or https://regex101.com/ to help understand regex.

counterexample (use one of the 2 websites), the RegExp is `code` and line below RegExp is the text string:
`\{.*\}`
fwefwefwef{very greedy  }    more cake}wefwefwefwef

`\{.*?\}`
fwefwefwef{nope, I've had enough  }    more cake}wefwefwefwef
___
I will have to find and understand the other use cases

Reply:
Gotcha. Then of course don't change the regex! I will say, this PR is a bit stale. I am looking into fixing the last January bug now (#18), hopefully I will learn a bit about the formatter